### ISSUE TITLE
[BUG] generator has created models which name is empty if title including unicode character

### ISSUE THREAD

**Issue by y-chan** [OpenAPITools/openapi-generator#15942]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

I have created TypeScript client code by OpenAPI Generator, but generator outputted invalid code.
IMHO, if title field including unicode character, it is caused.
Normally, we don't make model name which include unicode, but we make title which include unicode.
Because right name has as key, we use title as short description.
<img width="263" alt="image" src="https://github.com/OpenAPITools/openapi-generator/assets/34832037/d481472a-b315-4889-99c8-817b97dbdb2b">
<img width="463" alt="image" src="https://github.com/OpenAPITools/openapi-generator/assets/34832037/20537d41-bccc-4d3d-80bd-454547ea1a1f">

I debug details(most easy `print` debug :joy:), I found model name only consists underscores inside.
![image](https://github.com/OpenAPITools/openapi-generator/assets/34832037/309fb662-68e8-46e9-a630-5462fb25bd93)
I think it was converted camelCase, deleted underscores, so model name is empty.

##### openapi-generator version

6.6.0

##### OpenAPI declaration file content or url

<summary>Details </summary>
<details>

```json
{
   "openapi":"3.0.2",
   "info":{
      "title":"VOICEVOX Engine",
      "description":"VOICEVOXの音声合成エンジンです。",
      "version":"latest"
   },
   "paths":{
      "/accent_phrases":{
         "post":{
            "tags":[
               "クエリ編集"
            ],
            "summary":"テキストからアクセント句を得る",
            "description":"テキストからアクセント句を得ます。\nis_kanaが`true`のとき、テキストは次のようなAquesTalkライクな記法に従う読み仮名として処理されます。デフォルトは`false`です。\n* 全てのカナはカタカナで記述される\n* アクセント句は`/`または`、`で区切る。`、`で区切った場合に限り無音区間が挿入される。\n* カナの手前に`_`を入れるとそのカナは無声化される\n* アクセント位置を`'`で指定する。全てのアクセント句にはアクセント位置を1つ指定する必要がある。\n* アクセント句末に`？`(全角)を入れることにより疑問文の発音ができる。",
            "operationId":"accent_phrases_accent_phrases_post",
            "parameters":[
               {
                  "required":true,
                  "schema":{
                     "title":"Text",
                     "type":"string"
                  },
                  "name":"text",
                  "in":"query"
               },
               {
                  "required":true,
                  "schema":{
                     "title":"Speaker",
                     "type":"integer"
                  },
                  "name":"speaker",
                  "in":"query"
               },
               {
                  "required":false,
                  "schema":{
                     "title":"Is Kana",
                     "type":"boolean",
                     "default":false
                  },
                  "name":"is_kana",
                  "in":"query"
               },
               {
                  "required":false,
                  "schema":{
                     "title":"Core Version",
                     "type":"string"
                  },
                  "name":"core_version",
                  "in":"query"
               }
            ],
            "responses":{
               "200":{
                  "description":"Successful Response",
                  "content":{
                     "application/json":{
                        "schema":{
                           "title":"Response Accent Phrases Accent Phrases Post",
                           "type":"array",
                           "items":{
                              "$ref":"#/components/schemas/AccentPhrase"
                           }
                        }
                     }
                  }
               },
               "422":{
                  "description":"Validation Error",
                  "content":{
                     "application/json":{
                        "schema":{
                           "$ref":"#/components/schemas/HTTPValidationError"
                        }
                     }
                  }
               }
            }
         }
      }
   },
   "components":{
      "schemas":{
         "AccentPhrase":{
            "title":"AccentPhrase",
            "required":[
               "moras",
               "accent"
            ],
            "type":"object",
            "properties":{
               "moras":{
                  "title":"モーラのリスト",
                  "type":"array",
                  "items":{
                     "$ref":"#/components/schemas/Mora"
                  }
               },
               "accent":{
                  "title":"アクセント箇所",
                  "type":"integer"
               },
               "pause_mora":{
                  "title":"後ろに無音を付けるかどうか",
                  "allOf":[
                     {
                        "$ref":"#/components/schemas/Mora"
                     }
                  ]
               },
               "is_interrogative":{
                  "title":"疑問系かどうか",
                  "type":"boolean",
                  "default":false
               }
            },
            "description":"アクセント句ごとの情報"
         },
         "HTTPValidationError":{
            "title":"HTTPValidationError",
            "type":"object",
            "properties":{
               "detail":{
                  "title":"Detail",
                  "type":"array",
                  "items":{
                     "$ref":"#/components/schemas/ValidationError"
                  }
               }
            }
         },
         "Mora":{
            "title":"Mora",
            "required":[
               "text",
               "vowel",
               "vowel_length",
               "pitch"
            ],
            "type":"object",
            "properties":{
               "text":{
                  "title":"文字",
                  "type":"string"
               },
               "consonant":{
                  "title":"子音の音素",
                  "type":"string"
               },
               "consonant_length":{
                  "title":"子音の音長",
                  "type":"number"
               },
               "vowel":{
                  "title":"母音の音素",
                  "type":"string"
               },
               "vowel_length":{
                  "title":"母音の音長",
                  "type":"number"
               },
               "pitch":{
                  "title":"音高",
                  "type":"number"
               }
            },
            "description":"モーラ（子音＋母音）ごとの情報"
         }
      }
   }
}
```

</details>

##### Generation Details

I use this command.
```bash
java -jar -Dfile.encoding=UTF-8  modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -i openapi.json     -g typescript-fetch   -o test_output/  --additional-properties "modelPropertyNaming=camelCase,supportsES6=true,withInterfaces=true,typescriptThreePlus=true"
```

##### Steps to reproduce

1. copy and paste json at details.
2. execute command at `Generation Details`

##### Suggest a fix

I'll send pull request, it can fix.

**Comment by wing328** [comment]
> IMHO, if title field including unicode character, it is caused.

so if the title field is "アクセント箇所" for example, the correct fix is that the inline model resolver shouldn't use title but use the default naming methodology instead, right?

**Comment by y-chan** [comment]
resolved by #16035
